### PR TITLE
#604 | View allowances

### LINKED ISSUE #604: Allowances - View - API Integration
Add API integration to the Allowances table for readonly use (no revoking)

- **TODO what should cancelled allowances toggle do? hide allowances with None?** yes
- **TODO How do we know if allowance is 'unlimited' from the API? and none?** the allowance is unlimited if it is greater than 999 trillion (consider token decimals)
- **TODO add sorting for all table headers**
- because clientside filtering is needed, fetch all allowances data for the user on page load and save it to the store
- fetch erc20, erc721, and erc1155 approvals each with a 10,000 limit from `/account/{account}/approvals`
- data from the `contracts` and `results` objects should be merged so that there is a list of `approvals` keyed by owner address with the followiung information:
    - tokenLogo (string or undefined - undefined for NFTs and for erc20 tokens with no logo - can be gotten from token store)
    - assetAmount (BigNumber - the amount of the asset owned by the user in wei) 
    - decimals (number - decimals of the token contract)
    - symbol - (string or undefined - ticker symbol for erc20 tokens and NFTs which have it, gotten from the contracts object of the response)
    - nftName - (string or undefined - the name of the NFT, if approval is an NFT)
    - allowanceAmount - ('unlimited' or BigNumber - amount allowed in WEI)
    - fiatValue (number - amount in USD equivalent to all of the owned asset) 
    - nftId - (string or undefined - to be used in the Row component to fetch nft data) **TODO how to get this?**
    - contract (string) **TODO what does this represent in the allowances endpoint response? contract of the asset itself?**
    - spenderAddress (address)
    - spenderName (string or undefined - gotten by the getContract method from the chain store)
    - type ('token' | 'nft' - can be gotten from the contract `supportedInterfaces` field)
    - updated (number - ms from epoch)
- data should be refreshed on the page every 13 seconds
- filtering should work
    - filter by token: results should match if the text is part of a token's symbol (e.g. USDT), token name (e.g. Tether), or nft name
    - filter by allowance: "none" (match any case), "unlimited" (match any case), "0", or number exact match (consider token decimals here)
    - filter by spender: results should match if the entered text is either an exact match (when lowercased) or substring for the spender address (lowercase), or if the text is a substring (lowercase) of authorized spender name
    - filter by contract address: results should match if the contract address (lowercase) is an exact match or substring for the approval contract (lowercase)

### PR DESCRIPTION
**Pull request by ezra-sg**
# Fixes #604 

## Description
This PR enables viewing real allowance information in the UI (note that modifying allowances is not a part of it)

Note: the indexer does not currently return cancelled allowances, so the toggle for that is nonfunctional. The logic should be working but we will need to test that after the indexer is updated

Note: the indexer returns slightly malformed data for single ERC721 allowances, so they are not included in the table. they should appear when the indexer is updated

Note: in this PR, extra large allowances are called "huge". In the next PR this will be changed to "infinite" with additional logic

## Test scenarios
- run the code locally on mainnet: `git fetch --all && git checkout 604-view-allowances-api-integration && echo 'NETWORK="mainnet"' > .env && yarn dev`
- log in with the team account (`0x...eE37`)
- go to the allowances page
    - you should see a loading state, then-
    - you should see allowances sorted by asset quantity, descending
- use the pagination buttons at the bottom of the table to view the next page of allowances
    - you should see the next page of allowances
- change the rows per page value in the table controls
    - rows per page should change accordingly
- try various options for sorting via clicking (and using tab + space/enter) on the table headers
    - allowances should sort correctly
- type various search terms in the search bar
    - results should filter by search term (see the github issue for details on what should be filterable)

## Checklist:

-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have cleaned up the code in the areas my change touches
-   [x] My changes generate no new warnings
-   [ ] Any dependent changes have been merged and published in downstream modules
-   [x] I have checked my code and corrected any misspellings
-   [x] I have removed any unnecessary console messages
-   [x] I have included all english text to the translation file and/or created a new issue with the required translations for the currently supported languages
-   [x] I have tested for mobile functionality and responsiveness
-   [ ] I have added appropriate test coverage

### CODE REVIEW

File: src/antelope/chains/EVMChainSettings.ts [hunk 1]
```diff
@@ -382,8 +386,7 @@ export default abstract class EVMChainSettings implements ChainSettings {
 
         // we fix the supportedInterfaces property if it is undefined in the response but present in the request
         Object.values(response.contracts).forEach((contract) => {
-            contract.supportedInterfaces = contract.supportedInterfaces ||
-                params.type ? [params.type?.toLowerCase() as string] : undefined;
+            contract.supportedInterfaces = contract.supportedInterfaces || (params.type ? [params.type.toLowerCase()] : undefined);
```
Comment (ezra-sg) [inline src/antelope/chains/EVMChainSettings.ts:389]: fixes an issue caused by the lack of parens - `supportedInterfaces` was ending up like `[undefined]`

Comment (Viterbo) [reply src/antelope/chains/EVMChainSettings.ts:389]: nice catch!

**Review by Viterbo — APPROVED** [review]
Awesome work!!! :+1:

**Review by donnyquixotic — APPROVED** [review]

**Review by donnyquixotic — CHANGES_REQUESTED** [review]
I noticed some odd sorting when all column values are the same (value - "N/A", type - 'token'). Clicking on the sort arrows in the column header for columns that have the same value _does_ change the row order. This is either a bug, or it is sorting based on values that aren't displayed. If it __is__ sorting, we should be displaying the values, otherwise the sort function shouldn't do anything for those columns.

**Pull request merged**